### PR TITLE
implemented /api/reviews/:review_id and error handling functions for …

### PR DESCRIPTION
**Pull request by LSB616**
…400 and 404 paths

### CODE REVIEW

File: __tests__/controller_functions.test.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+const request = require("supertest");
+const db = require("../db/connection");
+const app = require("../app");
+const { string } = require("pg-format");
+const { checkIfReviewIdExists, isIdValid } = require('../controllers/controller_functions');
+
+
+describe('checkIfReviewIdExists', () => {
```
Comment (k-huggs) [inline __tests__/controller_functions.test.js:8]: You don't need to perform this, if there is no article your database will tell you by returning an empty array.

File: controllers/board-game-review-controllers.js [hunk 2]
```diff
@@ -24,3 +23,13 @@ exports.getReviews = (req, res, next) => {
     });
 };
 
+exports.getReview = (req, res, next) => {
+    const id = req.params.review_id
+    Promise.all([checkIfReviewIdExists(id), isIdValid(id), selectReview(id)])
```
Comment (k-huggs) [inline controllers/board-game-review-controllers.js:28]: You've added and extra connection to your db here which isn't completely necessary. You only want to check things are valid or exist when you need to. PSQL in this instance will tell you without having to open up other connections to dbs which can be an expensive process in time and cash

**Review by k-huggs — COMMENTED** [review]
Hi mate, you've added in the check for the reiview when you don't need to at this point

File: controllers/board-game-review-controllers.js (hunk 2 above)
Comment (LSB616) [reply controllers/board-game-review-controllers.js:28]: So should I implement a PSQL error handler instead?

**Pull request merged**